content/en/blog/2023/ambient-ebpf-redirection/index.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Leverage eBPF for traffic redirection in Istio ambient mode"
+description: An alternative approach to redirect application pod traffic to ztunnel in Istio ambient mode.
+publishdate: 2023-03-15
+attribution: "Iris Ding (Intel), Chun Li (Intel)"
+keywords: [istio,ambient,ztunnel,eBPF]
+---
+
+The istio-cni component running on each Kubernetes worker node is responsible for redirecting application pod traffics to ztunnel on that node. By default it relies on iptables and
+[Geneve](https://www.rfc-editor.org/rfc/rfc8926.html) tunnel to achieve this redirection.  Now a new approach which is based on eBPF is also available in Istio ambient mode for this purpose.
+
+## Why eBPF
+
+In the context of Istio ambient mode redirection, although performance is undoubtedly important, it is imperative to recognize that, programmability or possibility takes precedence. With eBPF, you can leverage additional context to make these changes in the kernel so that packets can bypass complex routing and simply arrive at their final destination, that is sought-after for redirection datapath. Moreover, eBPF will provide the same or better control in datapath compared with iptables while providing the end of user experience that we are looking for.
+
+## How it works
+
+The eBPF program attached to the [traffic control](https://man7.org/linux/man-pages/man8/tc-bpf.8.html) ingress and egress hook has been compiled into istio-cni component. The istio-cni component will watch pod events and attach/detach the eBPF program to related network interface when the pod is moved into/out of the ambient mode.

```

Comment:
This is confusing. It says that the eBPF program `is attached to the ...`, but then the next sentence says the istio-cni component will watch pod events and `attach/detach` the program. It sounds like this is saying that the same program is being attached to multiple things. Is that right? Would this be more clear?
```suggestion
An eBPF program, attached to the [traffic control](https://man7.org/linux/man-pages/man8/tc-bpf.8.html) ingress and egress hook, has been compiled into istio-cni component. The istio-cni component will watch pod events and attach/detach the eBPF program to other related network interfaces when the pod is moved into or out of ambient mode.

```

Reply:
Thanks for making it more clear. Updated